packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -3,470 +3,33 @@
  * Licensed under the MIT License.
  */
 
-import { IDisposable, ITelemetryLogger, ITelemetryProperties } from "@fluidframework/common-definitions";
-import { Deferred, IPromiseTimerResult, PromiseTimer, Timer } from "@fluidframework/common-utils";
+import { IDisposable, ITelemetryLogger } from "@fluidframework/common-definitions";
+import { Deferred, PromiseTimer } from "@fluidframework/common-utils";
 import {
     ISequencedDocumentMessage,
     ISequencedDocumentSystemMessage,
     ISummaryConfiguration,
     MessageType,
 } from "@fluidframework/protocol-definitions";
-import { ChildLogger, PerformanceEvent } from "@fluidframework/telemetry-utils";
+import { ChildLogger } from "@fluidframework/telemetry-utils";
 import {
-    GenerateSummaryData,
```

Comment:
Is there any way to break PR into "refactoring / no logical changes" and "everything else" PRs?

Reply:
I have split it into 3 separate PRs.
1. #6783 - refactor into summarizerTypes.ts
2. #6788 - refactor into summaryGenerator.ts
3. #6798 - add summarizeOnDemand